### PR TITLE
Expose the ArrayInput component

### PR DESCRIPTION
**Pull request by ellthompson**
Fixes #51 

Adds the ArrayInput component to the pcui build output and creates a set of Storybook stories for it.

Currently the ArrayInput component supports input types of: TextInput, NumericInput, BooleanInput & VectorInput (dimensions 2, 3 & 4).

image

### CODE REVIEW

File: src/components/ArrayInput/index.js [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+import Button from '../Button';
+
+const CLASS_ARRAY_INPUT = 'pcui-array-input';
+const CLASS_ARRAY_EMPTY = 'pcui-array-empty';
+const CLASS_ARRAY_SIZE = CLASS_ARRAY_INPUT + '-size';
+const CLASS_ARRAY_CONTAINER = CLASS_ARRAY_INPUT + '-items';
+const CLASS_ARRAY_ELEMENT = CLASS_ARRAY_INPUT + '-item';
+const CLASS_ARRAY_DELETE = CLASS_ARRAY_ELEMENT + '-delete';
+
+var DEFAULTS = {
```
Comment (vkalpias) [inline src/components/ArrayInput/index.js:17]: How are we going to support additional types is there a way to register new DEFAULTS for new types?

File: src/components/ArrayInput/index.js [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+ */
+
+/**
+ * @name ArrayInput
+ * @classdesc Element that allows editing an array of values.
+ * @property {boolean} renderChanges If true the input will flash when changed.
+ * @property {boolean} fixedSize If true then editing the number of elements that the array has will not be allowed.
+ * @property {object} elementArgs Arguments for each array Element.
+ * @property {object} inputClass The class that should be used as the input for each array element. Can be one of: TextInput, BooleanInput, AssetInput, VectorInput
+ * @augments Element
```
Comment (vkalpias) [inline src/components/ArrayInput/index.js:50]: We should also add the following:
`@mixes IBindable`
`@mixes IFocusable`

**Review by vkalpias — APPROVED** [review]
Approved with some comments

File: src/components/ArrayInput/index.js (hunk 1 above)
Comment (ellthompson) [reply src/components/ArrayInput/index.js:17]: I've had to change the way the array input class receives it's input element as there's no global element registration in pcui. Now the input class you want to use is passed in as an argument, along with the type string. We'll need to add in defaults here as new types are added. But it also means users wont be importing every input element class into their project every time they use the array input class.

File: src/components/ArrayInput/index.js (hunk 2 above)
Comment (ellthompson) [reply src/components/ArrayInput/index.js:50]: Cheers, i'll add these in 👍

**Pull request merged**